tests/CppUTest/UtestTest.cpp

```diff
@@ -218,9 +218,34 @@ TEST(UtestShell, TestStopsAfterSetupFailure)
 }
 
 #if CPPUTEST_USE_STD_CPP_LIB
+
+/*
+ * Prevents -Wunreachable-code; should always return true
+ */
+static bool avoidUnreachableCodeWarning()
+{
+    do
+    {
+        return true;
+    }
+    while(0);
+}
+
 static void thrownUnknownExceptionMethod_()
 {
-    throw 33;
+    if (avoidUnreachableCodeWarning())
+    {
+        throw 33;
+    }
+    stopAfterFailure++;
+}
+
+static void thrownStandardExceptionMethod_()
+{
+    if (avoidUnreachableCodeWarning())
```

Comment:
Nice job on tricking the compiler! :)

What I was thinking about was to not trick the compiler but to put a legitimate if there. It would be on a static-scoped boolean. We could then also add another test that actually sets the boolean to false thus not throwing. That way the unreachable code will actually be reached... and it will also not lead to non-coverage issues.

I guess it will be safer than the compiler trick as there will be a time the compiler (or another one) to notice the trick :)

Reply:
Using a non-const static seems enough. :+1:

Reply:
Yeah, non-const will export it as a linker symbol though, which isn't very nice. I'd prefer a const one and then having one tests where it is set to false. That will then also cover the 'unreachable' lines in the test coverage measurements, which I think it probably better. And then I will really merge it and move on to the next one!

Reply:
> I'd prefer a const one and then having one tests where it is set to false.

Changing the value of a `const` wont work (or do I get this wrong?).

Reply:
Yeah, so I would suggest to not make it const :)

Perhaps call it "shouldThrowException" and set it to true by default.

Reply:
Done